force-app/main/default/triggers/LeadTrigger.trigger

```diff
@@ -26,10 +26,32 @@ trigger LeadTrigger on Lead(before insert) {
 			LeadTriggerHandler.handleAutoLeadScoring(Trigger.new);
 		}
 		when AFTER_INSERT {
-			LeadTriggerHandler.handleLeadAutoConvert(Trigger.new);
+			// LeadTriggerHandler.handleLeadAutoConvert(Trigger.new); (Original Code)
+
+			// Check if lead's Email is populated;
+			List<Lead> leadsToAutoConvert = new List<Lead>();
+			for (Lead l : Trigger.new) {
+				if (l.Email != NULL) {
+					leadsToAutoConvert.add(l);
+				}
+			}
+			if (!leadsToAutoConvert.isEmpty()) {
+				LeadTriggerHandler.handleLeadAutoConvert(leadsToAutoConvert);				
+			}
 		}
 		when AFTER_UPDATE {
-			LeadTriggerHandler.handleLeadAutoConvert(Trigger.new);
+			// LeadTriggerHandler.handleLeadAutoConvert(Trigger.new); (Original Code)
+
+			// Check if lead's Email is updated AND IsConverted != TRUE
+			List<Lead> leadsToAutoConvert = new List<Lead>();
+			for (Lead l : Trigger.new) {
+				if (l.Email != Trigger.oldMap.get(l.Id).Email && l.IsConverted != TRUE) {
+					leadsToAutoConvert.add(l);
+				}
+			}
+			if (!leadsToAutoConvert.isEmpty()) {
+				LeadTriggerHandler.handleLeadAutoConvert(leadsToAutoConvert);				
```

Comment:
Try to remove all this logic out of the trigger and move it to the trigger handler. The original code is exactly what you want.